src/services/salesforce/connection.ts

```diff
@@ -33,18 +34,26 @@ export class SalesforceConnection extends Connection {
     const nameList = Array.from(sources.keys())
       .map(name => `'${name}'`)
       .join(', ');
-    const existingClasses = await this.tooling.query(
-      `Select Id From ApexClass where Name in (${nameList})`
+    const existingClasses = await retry(
+      async () =>
+        await this.tooling.query(
+          `Select Id From ApexClass where Name in (${nameList})`
+        )
```

Comment:
I think you can simplify to `await retry(() => promise)` ?

Reply:
The jsforce typing does not allow that, 

`Type 'Query<Schema, string, Record, \"QueryResult\">' is missing the following properties from type 'Promise<unknown>': finally, [Symbol.toStringTag]`

Looked to me like although you can await query, we are missing something to say the return is a Promise.

Reply:
Ah yes, I've come across that before - Query implements / pretends to be Promise (or Thenable?) somewhere in the typings so it doesn't always work. Shame.